jpf-symbc/src/main/gov/nasa/jpf/symbc/witness/Edge.java

```diff
@@ -0,0 +1,99 @@
+        // If variable value is null, it means that this variable doesn't appear in PC
+        // (i.e. it doesn't matters program's correctness. In this case, just put arbitrary default value)
+        if(symbolicVariableInfoList.get(indexOfEdge).returnType.equals("double") ||
+                symbolicVariableInfoList.get(indexOfEdge).returnType.equals("float")){
+            if(symbolicVariableInfoList.get(indexOfEdge).varValue == null) {
+                edgeBuilder.append(String.format("         <data key=\"assumption\">%s == %f</data>\n", symbolicVariableInfoList.get(indexOfEdge).varName, 2.0));
+            }
+            else edgeBuilder.append(String.format("         <data key=\"assumption\">%s == %f</data>\n", symbolicVariableInfoList.get(indexOfEdge).varName, symbolicVariableInfoList.get(indexOfEdge).varValue));
+        }
+
+        else if(symbolicVariableInfoList.get(indexOfEdge).returnType.equals("boolean")){
+            if(symbolicVariableInfoList.get(indexOfEdge).varValue == null) edgeBuilder.append(String.format("         <data key=\"assumption\">%s == %b</data>\n", symbolicVariableInfoList.get(indexOfEdge).varName, false));
+            else edgeBuilder.append(String.format("         <data key=\"assumption\">%s == %b</data>\n", symbolicVariableInfoList.get(indexOfEdge).varName, symbolicVariableInfoList.get(indexOfEdge).varValue));
+        }
+
+        else if(symbolicVariableInfoList.get(indexOfEdge).returnType.equals("java.lang.String")){
+            if(allowMethodInvocation){
+                if(symbolicVariableInfoList.get(indexOfEdge).varValue == null) edgeBuilder.append(String.format("         <data key=\"assumption\">%s.equals(%s)</data>\n", symbolicVariableInfoList.get(indexOfEdge).varName, "\"\""));
+                else edgeBuilder.append(String.format("         <data key=\"assumption\">%s.equals(%s)</data>\n", symbolicVariableInfoList.get(indexOfEdge).varName,symbolicVariableInfoList.get(indexOfEdge).varValue));
+            }
+            else{
+                if(symbolicVariableInfoList.get(indexOfEdge).varValue == null) edgeBuilder.append(String.format("         <data key=\"assumption\">%s==%s</data>\n", symbolicVariableInfoList.get(indexOfEdge).varName, "\"\""));
+                else edgeBuilder.append(String.format("         <data key=\"assumption\">%s==%s</data>\n", symbolicVariableInfoList.get(indexOfEdge).varName, symbolicVariableInfoList.get(indexOfEdge).varValue));
+            }
+        }
+
+        else{
+            // Use == to follow standard graphml format
+            if(symbolicVariableInfoList.get(indexOfEdge).varValue == null) edgeBuilder.append(String.format("         <data key=\"assumption\">%s == %d</data>\n", symbolicVariableInfoList.get(indexOfEdge).varName, 4));
+            else edgeBuilder.append(String.format("         <data key=\"assumption\">%s == %d</data>\n", symbolicVariableInfoList.get(indexOfEdge).varName, symbolicVariableInfoList.get(indexOfEdge).varValue));
+        }
```

Comment:
i think here you are expecting to see an integer value for your symbolic value, can you be more explicit about that. Can you have a comment that says this and perhaps an assert statement. Something like 
`assert returnType.equals("Integer")`

Reply:
Here, the type of symbolic variable could be integer or char(for character type, it is ok to write char type variable as numeric value. You can check this at GDart's result of SV-COMP(see jbmc-regression/StaticCharMethods04): [https://sv-comp.sosy-lab.org/2024/results/results-verified/gdart.results.SV-COMP24.table.html#/table)](https://sv-comp.sosy-lab.org/2024/results/results-verified/gdart.results.SV-COMP24.table.html#/table). 

The benchmark StaticCharMethods04 looks like this : 
```Java
import org.sosy_lab.sv_benchmarks.Verifier;

public class Main {
  public static void main(String[] args) {
    char c = Verifier.nondetChar();
    assert Character.isLetter(c);
  }
}
```

I added comments and assert statement that explicitly shows this. I'm removing this comments. Thank you!